errors.go

```diff
@@ -222,8 +231,11 @@ func IsNotImplemented(e error) bool {
 	type ni interface {
 		IsNotImplementedError() bool
 	}
-	err, ok := Unwrap(e).(ni)
-	return ok && err.IsNotImplementedError()
+
+	return recursiveIsError(e, func(e error) bool {
+		err, ok := e.(ni)
+		return ok && err.IsNotImplementedError()
```

Comment:
This was the only check that actually called the interface method. I've kept the original behavior.